lib/galaxy/model/__init__.py

```diff
@@ -1510,7 +1510,13 @@ def __init__( self, history, action, role ):
         self.role = role
 
 
-class Dataset( object ):
+class StorableObject( object ):
+
+    def __init__( self, id, **kwargs):
+        self.id = id
```

Comment:
Since `id` is not used for `MetadataFile`, I'd say that you can change this to a simple:
```
    def __init__( self ):
        pass
```
and restore `self.id = id` to its original position.